frontend/server/src/DAO/Users.php

```diff
@@ -447,4 +447,36 @@ final public static function validateDeletionToken(
 
         return boolval($count);
     }
+
+    public static function findByParentalToken(string $token): array {
+        $fields = join(
+            ', ',
+            array_map(
+                fn (string $field): string => "u.{$field}",
+                array_keys(
+                    \OmegaUp\DAO\VO\Users::FIELD_NAMES
+                )
+            )
+        );
+        $sql = "SELECT
+                      {$fields}
+
+                    FROM
+                        Users u
+                    INNER JOIN
+                        Identities i ON u.main_identity_id = i.identity_id
+                    WHERE
+                      parental_verification_token = ?
+                    LIMIT 1;";
```

Comment:
```suggestion
                    LIMIT 1
                    FOR UPDATE;";
```

not _entirely_ sure where the the [`FOR UPDATE`](https://dev.mysql.com/doc/refman/5.6/en/innodb-locking-reads.html) should go, though.